definitions/npm/@ginkgo-bioworks/react-json-schema-form-builder_v2.x.x/flow_v0.92.x-/react-json-schema-form-builder_v2.x.x.js

```diff
@@ -0,0 +1,110 @@
+declare module '@ginkgo-bioworks/react-json-schema-form-builder' {
+  declare type Parameters = {|
+    [string]: string | number | boolean | Array<string | number>,
+    name: string,
+    path: string,
+    definitionData: { [string]: any, ... },
+    definitionUi: { [string]: any, ... },
+    category: string,
+    'ui:option': { [string]: any, ... },
+  |};
+
+  declare export type CardBodyProps = {|
+    parameters: Parameters,
+    onChange: (newParams: Parameters) => void,
+  |};
+
+  declare type DataType =
+    | 'string'
+    | 'number'
+    | 'boolean'
+    | 'integer'
+    | 'array'
+    | '*'
+    | null;
+
+  declare type MatchType = {|
+    types: Array<DataType>,
+    widget?: string,
+    field?: string,
+    format?: string,
+    $ref?: boolean,
+    enum?: boolean,
+  |};
+
+  declare type FormInput = {|
+    displayName: string,
+    // given a data and ui schema, determine if the object is of this input type
+    matchIf: Array<MatchType>,
+    // allowed keys for ui:options
+    possibleOptions?: Array<string>,
+    defaultDataSchema: {
+      [string]: any,
+      ...
+    },
+    defaultUiSchema: {
+      [string]: any,
+      ...
+    },
+    // the data schema type
+    type: DataType,
+    // inputs on the preview card
+    cardBody: React$ComponentType<CardBodyProps>,
+    // inputs for the modal
+    modalBody?: React$ComponentType<CardBodyProps>,
+  |};
+
+  declare type DataOptions = {|
+    title: string,
+    type?: string,
+    description?: string,
+    $ref?: string,
+    default?: string,
+  |};
+
+  // optional properties that can add custom features to the form builder
+  declare type Mods = {|
+    customFormInputs?: {
+      [string]: FormInput,
+      ...
+    },
+    tooltipDescriptions?: {|
+      add?: string,
+      cardObjectName?: string,
+      cardDisplayName?: string,
+      cardDescription?: string,
+      cardInputType?: string,
+      cardSectionObjectName?: string,
+      cardSectionDisplayName?: string,
+      cardSectionDescription?: string,
+    |},
+    labels?: {|
+      formNameLabel?: string,
+      formDescriptionLabel?: string,
+      objectNameLabel?: string,
+      displayNameLabel?: string,
+      descriptionLabel?: string,
+      inputTypeLabel?: string,
+    |},
+    showFormHead?: boolean,
+    deactivatedFormInputs?: Array<string>,
+    newElementDefaultDataOptions?: DataOptions,
+  |};
+
+  declare type FormBuilderProps = {|
+    schema: string,
+    uischema: string,
+    onChange: (string, string) => any,
```

Comment:
Do you think it would be helpful to add the a var name so that as a user I don't need to look up documentation on what first and second strings do?
```suggestion
    onChange: (bah1: string, blah2: string) => any,
```

Reply:
Good suggestion -- done